### PR TITLE
Fix #1448 - Add hostname to Discord notifications

### LINKED ISSUE #1448: Discord notifications : the Service URL is empty for some tests
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

In the Discord notification, the Service URL is empty or only contain "https://" when the Monitor type is not HTTP.

image

### 👟 Reproduction steps

1. Setup a Discord Notification
2. Create a Ping Monitor with this notification enabled
3. Make it fail or succed

### 👀 Expected behavior

The Service URL should not be here.

### 😓 Actual Behavior

The Sevice URL field is empty or only contain "https://".

### 🐻 Uptime-Kuma Version

1.11.3

### 💻 Operating System and Arch

Debian 10

### 🌐 Browser

Firefox 98

### 🐋 Docker Version

_No response_

### 🟩 NodeJS Version

_No response_

### 📝 Relevant log output

https://github.com/louislam/uptime-kuma/blob/dd183e2ec25d74429541ca06c412d53c92e2cb92/server/notification-providers/discord.js#L50-L53

### PR DESCRIPTION
**Pull request by jordanbertasso**
# Description

Fixes #1448. There was a case for `port` types but not `ping`, `dns` and `steam` types, so I just reused the logic from `port` type monitors.

## Type of change

Please delete any options that are not relevant.

- Bug fix

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I ran ESLint and other linters for modified files
- [X] I have performed a self-review of my own code and tested it
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] My code needed automated testing. I have added them (this is optional task)

## Screenshots

Screen Shot 2022-04-10 at 9 49 59 pm
Screen Shot 2022-04-10 at 9 50 10 pm

### CODE REVIEW

File: server/notification-providers/discord.js [hunk 1]
```diff
@@ -24,14 +24,17 @@ class Discord extends NotificationProvider {
...
-                if (monitorJSON["port"]) {
-                    url += ":" + monitorJSON["port"];
-                }
-
-            } else {
-                url = monitorJSON["url"];
+            switch (monitorJSON["type"]) {
+                case "port":
+                case "ping":
+                    url = monitorJSON["hostname"];
```
Comment (Computroniks) [inline server/notification-providers/discord.js:30]: Doesn't this need to include all of the other none HTTP types such as `dns`?

**Review by Computroniks — CHANGES_REQUESTED** [review]

File: server/notification-providers/discord.js (hunk 1 above)
Comment (jordanbertasso) [reply server/notification-providers/discord.js:32]: They are covered now

**Comment by DasCanard** [comment]
Hey @jordanbertasso,
does this PR also fix #1678 ?

**Comment by karelkryda** [comment]
@jordanbertasso as mentioned above, it looks like you didn't included `push` type into this fix, did you?

**Comment by jordanbertasso** [comment]
Seems like it was missed @DasCanard @karelkryda

**Pull request merged**